lib/demon/base.rb

```diff
@@ -150,6 +150,7 @@ def run
     monitor_parent
     establish_app
     after_fork
+    exit 0 # Otherwise execution will return to wherever `start` was called
```

Comment:
this exit is really really confusing to me, it is exiting the process you just started

Reply:
`Demon#after_fork` is the function which contains the business logic for each demon. If `after_fork` finishes execution, it means the demon is dead. We need an exit here to avoid the demon turning into a unicorn master.

Reply:
I see, makes sense! lest move it to an ensure block and merge away!